.github/actions/prepare-api-variables/action.yml

```diff
@@ -0,0 +1,75 @@
+name: Prepare api variables
+description: Prepare env config for cloud
+
+inputs:
+  MODE:
+    description: "The mode of running tests (pull-request, release, main)"
+    required: true
+
+outputs:
+  BASE_URL:
+    description: "Dashboard base url"
+    value: ${{ steps.generate.outputs.BASE_URL }}
+  API_URL:
+    description: "API url"
+    value: ${{ steps.generate.outputs.API_URL }}
+  POOL_NAME:
+    description: "The name of the instance"
+    value: ${{ steps.generate.outputs.POOL_NAME }}
+  POOL_INSTANCE:
+    description: "The full URL of the instance"
+    value: ${{ steps.generate.outputs.POOL_INSTANCE }}
+  BACKUP_NAMESPACE:
+    description: "The backup namespace"
+    value: ${{ steps.generate.outputs.BACKUP_NAMESPACE }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Saleor login
+      uses: ./.github/actions/cli-login
+      with:
+        token: ${{ inputs.CLI_TOKEN }}
+
+    - name: Inject slug/short variables
+      uses: rlespinasse/github-slug-action@102b1a064a9b145e56556e22b18b19c624538d94
+
+    - name: Generate
+      id: generate
+      shell: bash
+      env:
+        MODE: ${{ inputs.MODE }}
+        PREFIX: pr-
+      run: |
+        if [[ $MODE == 'pull-request' ]]; then
+          echo "BASE_URL=${PREFIX}${GITHUB_HEAD_REF_SLUG_URL}.dashboard.saleor.rocks" >> $GITHUB_OUTPUT
+          echo "API_URL=https://${PREFIX}${GITHUB_HEAD_REF_SLUG_URL}.staging.saleor.cloud/graphql/" >> $GITHUB_OUTPUT
+          echo "POOL_NAME=${PREFIX}${GITHUB_HEAD_REF_SLUG_URL}" >> $GITHUB_OUTPUT
+          echo "POOL_INSTANCE=https://${PREFIX}${GITHUB_HEAD_REF_SLUG_URL}.staging.saleor.cloud" >> $GITHUB_OUTPUT
+          echo "BACKUP_NAMESPACE=snapshot-automation-tests" >> $GITHUB_OUTPUT
+          
+          exit 0
```

Comment:
You can use `case` or `elif` instead as well, this would be cleaner. The script is also missing a check that `$MODE` is a valid value as it only seem to expect `pull-request|release|main` but doesn't throw an error if it's not in that list.

Suggestion:

```sh
case "$MODE" in
pull-request)
  # [...]
  ;;

release)
  # [...]
  ;;

main)
  # [...]
  ;;
*)
  echo "Unknown mode: $MODE" >&2
  exit 1
  ;;
esac
```

Reply:
I know, but i would keep the if statements - i'm not sure if these conditions will stay just with string comparison